packages/fastify-renderer/src/node/DocumentTemplate.tsx

```diff
@@ -2,8 +2,8 @@ import template from 'stream-template'
 
 /** Data passed to the template function by the renderer */
 export interface TemplateData<Props> {
-  head: string
-  tail: string
+  head: NodeJS.ReadableStream
+  tail: NodeJS.ReadableStream
```

Comment:
👍 Makes sense to me! I think it may be a bit awkward at some points to have to convert all the stuff you might want to write into stream write calls, but if we're gonna do streaming, that is the deal!